docker/Dockerfile

```diff
@@ -47,38 +49,35 @@ RUN ln -sf /usr/bin/python3 /usr/bin/python
 # -------------=== python build ===-------------
 
 #Build python intermediate docker image
-FROM ubuntu:bionic as python_image
+FROM $IMAGE as python_image
+ARG DISTRO
 
+COPY ./scripts/install_packages /project/avalon/scripts/
 
 # Add necessary packages
-RUN apt-get update \
- && apt-get install -y -q \
-    ca-certificates \
-    pkg-config \
-    python3-pip \
-    python3-dev \
-    make \
- && apt-get clean
-
-# Install setuptools packages using pip because
-# these are not available in apt repository.
-RUN pip3 install setuptools
-
+RUN packages="ca-certificates pkg-config python3-pip make" \
+    pip_packages="setuptools"; \
+    if [ "$DISTRO" = "bionic" ] ; then \
+      packages="$packages python3-dev"; \
+    elif [ "$DISTRO" = "centos" ] ; then \
+      packages="$packages python3-devel" \
+      pip_packages="$pip_packages wheel"; \
```

Comment:
Please use packages="$packages python3-wheel".  Let's use dnf by dfeault. If package is not available with dnf then only use pip.

Reply:
I would prefer to take this change as next level of optimization as part of seperate PR.

Reply:
This is not an optimization issue. There can be conflict (eg: dependent library versions) if we mix and match pip and dnf/yum/apt-get.  So it can impact functionality. The recommended approach is to consistently use one package manger by default. Let's say we use dnf in case of centos 8 and use pip only if that specific package if not available with dnf.

Reply:
okay. Let me check this.